initialization/initialization.go

```diff
@@ -435,6 +441,17 @@ func (init *Initializer) initFile(ctx context.Context, fileIndex int, batchSize,
 		init.logger.Info("initialization: starting to write file #%v; target number of labels: %v, start position: %v", fileIndex, fileNumLabels, fileOffset)
 	}
 
+	wo, err := oracle.New(
+		oracle.WithProviderID(uint(init.opts.ProviderID)),
+		oracle.WithCommitment(init.commitment),
+		oracle.WithVRFDifficulty(difficulty),
+		oracle.WithScryptParams(init.opts.Scrypt),
+	)
+	if err != nil {
+		return err
+	}
```

Comment:
Primarily because I thought it would make the process to cancel initialization more difficult, but now I'm not sure if it would be.

I would not add the work oracle to the state of the initializer (I eventually want to get rid of the service and just replace it with a function), but maybe I can instantiate a WorkOracle at the start of the function and close it when it returns instead of once per file...